src/ProductDetails.js

```diff
@@ -0,0 +1,40 @@
+import useFetch from "./hooks/useFetch";
+import Loading from "./components/Loading"
+import { useParams, useNavigate } from "react-router-dom";
+
+const ProductDetails = (props) => {
+    const {productId} = useParams();
+    const {data, loading, error} = useFetch('products');
+    const product = data[productId - 1];
+    const navigate = useNavigate();
+
+    const handleBackButtonClick = () => {
+        navigate(`/product-page`);
```

Comment:
`navigate(-1)` will allow you to go back to whatever the previous page was. This would be useful if we were directly navigating to products/1 and then typing in products/2. While on products/2, your implementation would still go back to the product page but with `navigate(-1)` we'd go back to products/1

Reply:
Thanks for the tip! made the change in my program.